### PR TITLE
save engine to deployment_hosts

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
@cfchase, please test this. It was easier to create another route / method and use the controller rather than a model method using the PUT update action.

File: app/controllers/fusor/api/v21/deployments_controller.rb [hunk 1]
```diff
@@ -384,6 +385,38 @@ def openshift_disk_space
...
+  def add_rhv_engine
+    engine_id = params['engine_id']
+    engine_name = params['engine_name']
+
+    # don't delete/add rows if nothing changed
+    if (@deployment.rhev_engine_host_id != engine_id) ||
+       (@deployment.rhev_self_hosted_engine_hostname != engine_name)
+
+      rhv_engine = DeploymentHost.where(deployment_id: @deployment.id,
+                           discovered_host_id: @deployment.rhev_engine_host_id,
```
Comment (cfchase) [inline app/controllers/fusor/api/v21/deployments_controller.rb:397]: searching for a discovered_host_id: nil always yielded no rhv engine.  Omitting discovered_host_id fixes it:

```ruby
rhv_engine = DeploymentHost.where(
  deployment_id: @deployment.id, 
  deployment_host_type: 'rhv_engine'
).first
```

**Comment by cfchase** [comment]
tested with that 1 change with https://github.com/fusor/fusor-ember-cli/pull/10 and everything worked

**Comment by cfchase** [comment]
ACK

**Pull request merged**